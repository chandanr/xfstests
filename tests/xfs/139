#! /bin/bash
# SPDX-License-Identifier: GPL-2.0
# Copyright (c) 2015, Oracle and/or its affiliates.  All Rights Reserved.
#
# FS QA Test No. 139
#
# Try to ENOSPC while expanding the refcntbt by CoWing every block
# of a file that eats the whole AG.
#
seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1    # failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
    cd /
    rm -rf $tmp.* $testdir
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/reflink

# real QA test starts here
_supported_os Linux
_require_scratch_reflink
_require_cp_reflink

rm -f $seqres.full

_scratch_mkfs >/dev/null 2>&1
_scratch_mount
blksz=$(_get_file_block_size $SCRATCH_MNT)
_scratch_unmount

echo "Format and mount"
_scratch_mkfs -d agsize=$((8192 * $blksz)) > $seqres.full 2>&1
_scratch_mount >> $seqres.full 2>&1

testdir=$SCRATCH_MNT/test-$seq
mkdir $testdir

echo "Create the original files"
sz=$((12288 * $blksz))
_pwrite_byte 0x61 0 $sz $testdir/file1 >> $seqres.full
_cp_reflink $testdir/file1 $testdir/file2 >> $seqres.full
_scratch_cycle_mount

echo "CoW every other block"
_pwrite_byte 0x62 0 $sz $testdir/file1 >> $seqres.full
_scratch_cycle_mount

echo "Compare files"
od -t x1 $testdir/file1 | _filter_od
od -t x1 $testdir/file2 | _filter_od
#filefrag -v $testdir/file1 $testdir/file2

# success, all done
status=0
exit
