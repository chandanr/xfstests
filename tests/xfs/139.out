QA output created by 139
Format and mount
Create the original files
CoW every other block
Compare files
0 62 62 62 62 62 62 62 62 62 62 62 62 62 62 62 62
*
11300
0 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61
*
11300
